src/devices/Mcp3Adc/Mcp3004.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Device.Spi;
+
+namespace Iot.Device.Adc
+{
+    /// <summary>
+    /// MCP3004 Analog to Digital Converter (ADC)
+    /// </summary>
+    public class Mcp3004 : Mcp30xx32xx
+    {
+        /// <summary>
+        /// Constructs Mcp3004 instance
+        /// </summary>
+        /// <param name="spiDevice">Device used for SPI communication</param>
+        public Mcp3004(SpiDevice spiDevice) : base(spiDevice, pinCount: 4) { }
+
+        /// <summary>
+        /// Reads a 10-bit (0..1023) value from the device using pseudo-differential inputs
+        /// </summary>
+        /// <param name="channel">Channel which value should be read from (valid values: 0-1). Channel 0 is on inputs 0 and 1 and
+        /// Channel 1 is on inputs 2 and 3.</param>
+        /// <param name="polarityInverted">When false then then inputs 0 and 2 are positive and inputs 1 and 3 are negative
+        /// when true the polarity is swapped so that inputs 0 and 2 are negative and inputs 1 and 3 positive.</param>
+        /// <returns>10-bit value corresponding to relative voltage level on specified device channel.</returns>
+        public int ReadPseudoDifferential(int channel, bool polarityInverted = false)
```

Comment:
would it make sense to have something like `ReadDifferential(int referenceChannel, int channel)` and then if hardware has a way of representing this combination then use that, if not then just subtract two reads (perhaps add flag like forceHardwareImplementation which would cause it to throw if such is not available)

Reply:
@krwq
Makes perfect sense. I don't think that we need to 'force' the matter as the primary classes will know if the hardware supports true differential (which is the 3300 series I believe). Perhaps the channels should be named positiveChannel and negativeChannel as this fits in with the datasheet description. I also think that we should use the same two channel approach for ReadPseudoDifferential. Please give me a few days and I'll make these changes.

Reply:
@krwq
OK I have done a good deal of refactoring, added the 33xx family and other work.

 I have added support for using positive and negative channels. On the 32xx and 30xx family then Read differential will always just subtract values as hardware support is not available. On the 33xx family then ReadDifferential  checks  to see if the supplied channels exist and are in a valid pair and if they are then the hardware is used otherwise the values are subtracted like on the 32xx and 30xx series.

I have copied over the sample from the original Mcp3008 binding but I have not yet deleted Mcp3008 as there are other samples that are using that binding and before going ahead I wanted to be sure I am on the right track.

I haven't tested the 33xx code as I don't have any hardware samples but they should be with me real soon.